acceptance/tests/resource/service/launchd_provider.rb

```diff
@@ -79,4 +79,19 @@ def launchctl_assert_status(host, service, expect_running)
   ensure_service_on_host(agent, svc, {:enable => 'true'}) do
     launchctl_assert_status(agent, svc, false)
   end
+
+  # switching from stopped to running should output the correct status of the service and not 'absent'
+  step "Start the service on #{agent} when service is stopped, and check outoput"
+  on agent, puppet_resource('service', 'com.puppetlabs.sloth', 'ensure=stopped')
+  on agent, puppet_resource('service', 'com.puppetlabs.sloth', 'ensure=running', 'enable=true') do |result|
+    assert_match(/service { '#{svc}':\n  ensure => 'running',\n  \w+ => '\w+',\n}$/, stdout, 'Service status change failed')
+  end
+
+   # switching from running to stopped should output the correct status of the service and not 'absent'
+  step "Start the service on #{agent} when service is running, and check outoput"
```

Comment:
You should wrap the test in a do-end block.
Check the guidelines here: https://github.com/puppetlabs/beaker/blob/master/docs/concepts/style_guide.md